src/autoscaler/db/sqldb/appmetric_sqldb.go

```diff
@@ -83,35 +83,38 @@ func (adb *AppMetricSQLDB) SaveAppMetricsInBulk(appMetrics []*models.AppMetric)
 		stmt, err := txn.Prepare(pq.CopyIn("app_metric", "app_id", "metric_type", "unit", "timestamp", "value"))
 		if err != nil {
 			adb.logger.Error("failed-to-prepare-statement", err)
-			txn.Rollback()
+			_ = txn.Rollback()
 			return err
 		}
 		for _, appMetric := range appMetrics {
 			_, err := stmt.Exec(appMetric.AppId, appMetric.MetricType, appMetric.Unit, appMetric.Timestamp, appMetric.Value)
 			if err != nil {
 				adb.logger.Error("failed-to-execute", err)
-				txn.Rollback()
+				_ = txn.Rollback()
 				return err
 			}
 		}
 		_, err = stmt.Exec()
 		if err != nil {
 			adb.logger.Error("failed-to-execute-statement", err)
-			txn.Rollback()
+			_ = txn.Rollback()
 			return err
 		}
 		err = stmt.Close()
 		if err != nil {
 			adb.logger.Error("failed-to-close-statement", err)
-			txn.Rollback()
+			_ = txn.Rollback()
 			return err
 		}
 
 	case "mysql":
 		sqlStr := "INSERT INTO app_metric(app_id,metric_type,unit,timestamp,value)VALUES"
 		vals := []interface{}{}
 		if len(appMetrics) == 0 {
-			txn.Rollback()
+			err = txn.Rollback()
```

Comment:
not sure what this is about ?
I think you can just put _ = ?

Reply:
I've updated it to `return err`